x-pack/plugins/alerting/server/lib/process_alerts.ts

```diff
@@ -124,11 +124,12 @@ function processAlertsHelper<
           // this alert did exist in previous run
           // calculate duration to date for active alerts
           const state = existingAlerts[id].getState();
+          const currentState = activeAlerts[id].getState();
```

Comment:
This fixes a bug where an ongoing alert that uses `replaceState` will have its updated state value overriden.